common/src/main/java/tachyon/ClientBase.java

```diff
@@ -188,4 +191,90 @@ protected synchronized InetSocketAddress getAddress() {
     return mAddress;
   }
 
+  /**
+   * The RPC to be executed in {@link #retryRPC(RpcCallable)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallable<V> {
+    /**
+     * The task where RPC happens.
+     *
+     * @return RPC result
+     * @throws TException when any exception defined in thrift happens
+     */
+    V call() throws TException;
+  }
+
+  /**
+   * Same with {@link RpcCallable} except that this RPC call throws {@link TachyonTException} and
+   * is to be executed in {@link #retryRPC(RpcCallableThrowsTachyonTException)}.
+   *
+   * @param <V> the return value of {@link #call()}
+   */
+  protected interface RpcCallableThrowsTachyonTException<V> {
```

Comment:
I thought about this too. But not all Thrift RPC will throw these two exceptions, for example, those getters, and if we make such an assumption, then it will complicate client side exception handling because if an RPC throws a `TachyonTException`, the caller may need to ignore it(e.x. the getters that do not actually throw `TachyonTException`) or `unwrap` it to a more specific `TachyonException`.